### ISSUE TITLE
Is app.setVersion safe to use?

### ISSUE THREAD

**Issue by davidnorthetal** [electron/electron#13369]
* Electron version: 2.0.2

I saw there's `app.setVersion()` that I can use during runtime to set the version returned by `app.getVersion()` - however setVersion is not part of the official documentation and I suspected I have no guarantee that it's there and behaves like it does now tomorrow.

Can someone give me feedback if my assumption is correct, or do you think app.setVersion() is here to stay?

Thanks for any response.

**Comment by baconbrad** [comment]
It is basically how `app.getName()` works. It just overrides it. It is likely omitted to discourage it's use but it is safe to use as long as you know what you are doing for it's intended purpose. Just keep in mind it only overrides the version. It doesn't ensure Electron is using code and files for a particular version.

**Comment by miniak** [comment]
@davidnorthetal why do you need to use this API?

**Comment by davidnorthetal** [comment]
Hey, I evaluated using setVersion to set a non-semver compatible version-string (which can't be done via the version-field in `package.json`, as that is forced to be semver-compliant), but realized quickly that that was neither what code expects app.getVersion to do nor is it actually possible as how getVersion is currently implemented.

TL;DR I thought I could abuse it for something, turns out that's not a valid use case tho
